diff --git a/node_modules/pino-roll/lib/utils.js b/node_modules/pino-roll/lib/utils.js
index c84cd8f..0be7f5e 100644
--- a/node_modules/pino-roll/lib/utils.js
+++ b/node_modules/pino-roll/lib/utils.js
@@ -62,14 +62,19 @@ function getNext (frequency) {
   return getNextCustom(frequency)
 }
 
-function buildFileName (fileName, lastNumber = 1, extension) {
-  if (!fileName) {
+function getFileName (fileVal) {
+  if (!fileVal) {
     throw new Error('No file name provided')
   }
-  return `${fileName}.${lastNumber}${extension ?? ''}`
+  return typeof fileVal === 'function' ? fileVal() : fileVal
 }
 
-async function detectLastNumber (fileName, time = null) {
+function buildFileName (fileVal, lastNumber = 1, extension) {
+  return `${getFileName(fileVal)}.${lastNumber}${extension ?? ''}`
+}
+
+async function detectLastNumber (fileVal, time = null) {
+  const fileName = getFileName(fileVal)
   try {
     const numbers = await readFileTrailingNumbers(dirname(fileName), time)
     return numbers.sort((a, b) => b - a)[0]
@@ -102,4 +107,4 @@ async function isMatchingTime (filePath, time) {
   return birthtimeMs >= time
 }
 
-module.exports = { buildFileName, detectLastNumber, parseFrequency, getNext, parseSize }
+module.exports = { buildFileName, detectLastNumber, parseFrequency, getNext, parseSize, getFileName }
diff --git a/node_modules/pino-roll/pino-roll.js b/node_modules/pino-roll/pino-roll.js
index 800d59f..68c42e9 100644
--- a/node_modules/pino-roll/pino-roll.js
+++ b/node_modules/pino-roll/pino-roll.js
@@ -3,12 +3,19 @@
 const SonicBoom = require('sonic-boom')
 const { buildFileName, detectLastNumber, parseSize, parseFrequency, getNext } = require('./lib/utils')
 
+/**
+ * A function that returns a string path to the base file name
+ *
+ * @typedef {function} LogFilePath
+ * @returns {string}
+ */
+
 /**
  * @typedef {object} Options
  *
- * @property {string} file - Absolute or relative path to the log file.
+ * @property {string|LogFilePath} file - Absolute or relative path to the log file.
  * Your application needs the write right on the parent folder.
- * Number will be appened to this file name.
+ * Number will be appended to this file name.
  * When the parent folder already contains numbered files, numbering will continue based on the highest number.
  * If this path does not exist, the logger with throw an error unless you set `mkdir` to `true`.
  *
